### PR TITLE
#48: Fix ScalaDoc cross-module links

### LINKED ISSUE #48: Fix ScalaDoc cross-module links
## Background
Some _ScalaDoc_ cross-module links don't resolve correctly (at all)

## Feature
Fix _ScalaDoc_ cross-module links. At the time of issues creation there has been 4 warnings in this regard.
See: `sbt doc` output.

### PR DESCRIPTION
**Pull request by benedeki**
* fixed ScalaDoc cross-links
* decreased log size on tests running on Spark 3

Closes #48

### CODE REVIEW

File: spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala [hunk 1]
```diff
@@ -49,21 +48,45 @@ object DataFrameImplicits {
...
+     * @return  The string representation of the data in the DataFrame
+     * @since 0.2.0
+     */
     def dataAsString(numRows: Int, truncate: Boolean): String = {
       val showFnc: ()=>Unit = () => df.show(numRows, truncate)
       gatherData(showFnc)
     }
 
+    /**
+     * Get the string representation of the data in the format as [[org.apache.spark.sql.Dataset#show(numRows:Int,truncate:Int):Unit* spark.Dataset.show(numRows: Int, truncate: Int)]] displays them
```
Comment (lsulak) [inline spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:74]: Why all those `#` there? I couldn't find any reference to it in https://github.com/ThoughtWorksInc/sbt-api-mappings, and it seems to be working fine with `.` - including 'click to go to the definition' in the IntelliJ and `sbt doc` also seems to be producing no warnings

Comment (lsulak) [reply spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:74]: It even has Spark Mapping: https://github.com/ThoughtWorksInc/sbt-api-mappings/blob/5b1e4f11647421d287d3a26ed7fb077474ea7a3e/src/main/scala/com/thoughtworks/sbtApiMappings/SparkApiMappingRule.scala#L22

File: spark-commons/src/test/scala/za/co/absa/spark/commons/SparkVersionGuardTest.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  test("Guard with lower and without upper bounds") {
+    val guard = SparkVersionGuard(semver"2.0.0", None)
+
+    intercept[AssertionError] {
+      guard.ensureSparkVersionCompatibility("1.5.0")
+    }
+    guard.ensureSparkVersionCompatibility("2.0.0")
+    guard.ensureSparkVersionCompatibility("3.0.0")
+  }
+}
```
Comment (TebaleloS) [inline spark-commons/src/test/scala/za/co/absa/spark/commons/SparkVersionGuardTest.scala:42]: Please add a new line as the last line of this file

File: spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala (hunk 1 above)
Comment (miroslavpojer) [reply spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:74]: All my magic power comes from this page and one longer method header. ([See](https://github.com/scala/scala/blob/2.12.x/test/scaladoc/resources/links.scala))

Comment (lsulak) [reply spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:74]: Hmm okay, it's just a code docs, but it is from Scala 🤔 

well, I still think that we should use `.` primarily if it works: a) the destination is clickable and b) the `sbt doc` would be okay with it. And I think that many of these `#` can be replaced by `.` based on what I tried. I'll leave it up to you though

**Review by lsulak — APPROVED** [review]
LGTM

File: spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala [hunk 3]
```diff
@@ -37,9 +37,8 @@ object DataFrameImplicits {
       dfData
     }
 
-    //TODO  Fix ScalaDoc cross-module links #48 - Dataset.show()
     /**
-     * Get the string representation of the data in the format as Dataset.show() displays them
+     * Get the string representation of the data in the format as [[org.apache.spark.sql.Dataset#show():Unit* spark.Dataset.show()]] displays them
```
Comment (miroslavpojer) [inline spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:41]: This row do not navigate to defined method. Only to class.

Comment (benedeki) [reply spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataFrameImplicits.scala:41]: No idea how to improve on it. 🤷‍♂️ 
Once released, I want to see, if the link works in code navigation. 🤞

**Pull request merged**